app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventEditManagementActivity.kt

```diff
@@ -0,0 +1,197 @@
+                        Database.currentDatabase.eventDatabase!!.updateEvent(
+                            event
+                        ).observeOnce(this) {
+                            acceptEventEditCallback(it.value, event)
+                        }
+                    }
+                }else{
+                    HelperFunctions.showToast("The activity edit cannot be requested", this)
+                }
+            }
+
+            Unit
+        } else {
+            HelperFunctions.showToast("Can not accept this request", this)
+        }
+    }
+
+    private val declineEventEdit = { event: Event ->
+        createRefusalPopup(event)
+    }
+
+    private fun createRefusalPopup(event: Event) {
+        // Initialize a new layout inflater instance
+        val inflater: LayoutInflater =
+            getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+
+        // Inflate a custom view using layout inflater
+        val view = inflater.inflate(R.layout.popup_refuse_request, null)
+
+        // Initialize a new instance of popup window
+        val popupWindow = PopupWindow(
+            view, // Custom view to show in popup window
+            LinearLayout.LayoutParams.MATCH_PARENT, // Width of popup window
+            LinearLayout.LayoutParams.WRAP_CONTENT // Window height
+        )
+
+        val slideIn = Slide()
+        slideIn.slideEdge = Gravity.TOP
+        popupWindow.enterTransition = slideIn
+
+        // Slide animation for popup window exit transition
+        val slideOut = Slide()
+        slideOut.slideEdge = Gravity.END
+        popupWindow.exitTransition = slideOut
+
+
+        // Get the widgets reference from custom view
+        val confirmButton = view.findViewById<Button>(R.id.id_btn_confirm_refuse_request)
+        val message = view.findViewById<TextView>(R.id.id_txt_refusal_explanation)
+
+        //set focus on the popup
+        popupWindow.isFocusable = true
+
+        // Set a click listener for popup's button widget
+        confirmButton.setOnClickListener {
+            event.status = Event.EventStatus.REFUSED
+            event.adminMessage = message.text.toString()
+
+            Database.currentDatabase.eventDatabase!!.updateEventEdit(
+                event
+            ).observeOnce(this) {
+                if (!it.value) {
+                    HelperFunctions.showToast("Failed to decline the request", this)
+                } else {
+                    eventEdits.set(
+                        eventEdits.indexOfFirst { it2 -> it2.eventEditId == event.eventEditId },
+                        event,
+                        this
+                    )
+                }
+            }
+            // Dismiss the popup window
+            popupWindow.dismiss()
+        }
+
+        // Finally, show the popup window on app
+        TransitionManager.beginDelayedTransition(this.recyclerView)
+        popupWindow.showAtLocation(this.recyclerView, Gravity.CENTER, 0, 0)
+    }
```

Comment:
We can give informative feedback about why a request is not accepted, but we can not see it on the side of the activity provider

Reply:
Yes, if time, we will add the ability for the event organiser/activity provider to see its event edit requets. I already put the rejection message so that it will be easier to implement the functionality.